### DISCUSSION TITLE
Hide options in OptionLists

### DISCUSSION THREAD

**Discussion by rangerjo** [Textualize/textual#2313] [Ideas]
I would like to hide individual options in a `OptionList`. I'm trying to implement a fuzzy finder using an `Input` and a `OptionList`. 
Clearing all options using `clear_options()` and only adding the matching ones using `add_option()` works, but take a lot of time on larger option lists. At the same time, only disabling the options on input change is very quick. I've tried to set the `visibility: hidden` attribute on the component class `.option-list--option-disabled`, however the options do not disappear.
I would like to see a *hidden* feature for options in an `OptionList`. How would it be implemented? Extend the CSS functionality or add a new attribute to the `Option` class?

**Comment by davep** [comment]
>  I've tried to set the visibility: hidden attribute on the component class .option-list--option-disabled, however the options do not disappear.

This is likely something we need to expand on in the documentation at some point, but `.option-list--option-disabled` is what's known as a [component class](https://textual.textualize.io/guide/widgets/#component-classes); they can be styled from CSS but only a subset of styles are available -- mainly only do do with things like colour and text styles. Visibility isn't something that can be controlled with that.

**Comment by davep** [comment]
To address the main question here: while a filtering list is something we can imagine adding to Textual, it's unlikely that `OptionList` is where we'd do it. Because of the way `OptionList` works and the "hiding" would be no more faster than rebuilding the list from scratch again, as you're doing (`OptionList` is designed for an optimised for drawing lots of renderables of varying sizes).

Meanwhile, you may find something like [`textual-autocomplete`](https://github.com/darrenburns/textual-autocomplete) interesting; either as the widget to use, or as a source from which to build your own widget.
